07.ls_object/file_display.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+class FileDisplay
+  COLUMN_COUNT = 3
+
+  def initialize
+    @options = ARGV.getopts('alr').transform_keys(&:to_sym)
+    @file_details = fetch_file_details
+  end
+
+  def show_formatted_file_details
+    @options[:l] ? long_format : short_format
+  end
+
+  private
+
+  def fetch_file_details
+    flags = @options[:a] ? File::FNM_DOTMATCH : 0
+    file_names = Dir.glob('*', flags)
+    sorted_file_names = @options[:r] ? file_names.reverse : file_names
+    sorted_file_names.map { |file_name| FileDetail.new(file_name) }
+  end
+
+  def long_format
+    blocks_total = @file_details.map(&:blocks).sum
+    puts "total #{blocks_total}"
+
+    @file_details.each { |file_detail| puts format_file_detail(file_detail) }
+  end
+
+  def format_file_detail(file_detail)
+    max_size_map = build_max_size_map(@file_details)
```

Comment:
```suggestion
    max_sizes = build_max_sizes(@file_details)
```

ハッシュを複数形にすることもあるので、こっちの方がシンプルかも？

Reply:
こちら修正しました！